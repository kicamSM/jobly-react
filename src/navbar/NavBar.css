nav {
    background-color: black;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);
    /* height: 70px; */
    height: 150px;
  }
  
  /* .Nav {
    flex-direction: row;
    justify-content: center;
  } */

  /* .Nav {
    display: flex;
    justify-content: center;
  } */

  /* .nav-item {
    width: 100px;
    text-align: center;
    flex-direction: row;
  } */

    .nav-item {
    width: 100px;
    text-align: center;
    flex-direction: row;
  }
  
  nav a {
    color: #f0f2f5;
    text-decoration: none;
  }
  
  nav a.active {
    color: #f0f2f5;
    font-weight: bold;
    /* border-bottom: 5px dotted; */
    box-sizing: border-box;
  }

  /* .Navbar-Login {
    color: #f0f2f5;
    font-weight: bold;
    box-sizing: border-box;
  } */

  /* nav link.active {
    color: #f0f2f5;
    font-weight: bold;
    border-bottom: 5px dotted;
    box-sizing: border-box;
  }
   */
  nav a.active:hover {
    color: #f0f2f5 !important;
  }
  
  nav a:hover {
    text-decoration: none;
    color: #f0f2f5;
  }
  
  nav a.active.navbar-brand {
    border-bottom: none;
  }
  
  nav a:not(.navbar-brand):hover {
    font-weight: bold;
  }
  
 /* nav a.active.Navbar-Jobly {
    background-color: black;
    border-bottom: none;
  } */

  /* .Navbar-Jobly {
    margin-right: 50px;
  } */

  /* .Navbar-Companies {
   margin-left: 80vh;
  }

  .Navbar-Jobs {
    margin-left: 80vh;
   }

   .Navbar-Profile {
    margin-left: 80vh;
   }

   .Navbar-Logout {
    margin-left: 80vh;
   }  */

   /* .Nav {
    display: flex;
    flex-direction: row;
    justify-content: center;
  }

  .Navbar-Logout {
    background-color: black;
  } */

  
  /* .Navbar-button {
    background-color: black;
    color: white;
    
  } */

  /* .Navbar-button {
    color: inherit;
    text-decoration: none;
    border: none;
    cursor: pointer;
  }
  
  .Navbar-button:hover {
    background-color: #ccc;
  } */
