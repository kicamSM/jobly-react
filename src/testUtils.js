import React from "react";
import UserContext from "./repeated/UserContext";

const demoUser = {
  username: "testuser",
  firstName: "testfirst",
  lastName: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, user = demoUser, apply = () => false }) => (
    <UserContext.Provider value={{ user, apply }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
