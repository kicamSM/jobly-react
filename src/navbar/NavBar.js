import React from "react";
import { Link, NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import "./NavBar.css";


function NavBar({logout, user}) {
  console.log("user:", user)
  // const { firstName } = user
  // console.log("firstName:", firstName)

  if(user) {
   const { firstName } = user
    console.log("firstName:", firstName)
  }


  // console.log("user in nav", user.firstName)
  // !taking break will have to come back to getting user information out
  // !also note prefilling login page would be a good idea

// todo: insert name into logout. 

  return (
    <div> 
      <Navbar expand="md">
        <NavLink exact to="/" className="navbar-brand">
        </NavLink>
        <Nav className="ml-auto" navbar >
          {user && <> (
            <NavItem className="Navbar-Jobly">
              <NavLink to="/">Jobly</NavLink>
            </NavItem>
            <NavItem className="Navbar-Companies">
              <NavLink to="/companies">Companies</NavLink>
            </NavItem>
            <NavItem className="Navbar-Jobs">
              <NavLink to="/jobs">Jobs</NavLink>
            </NavItem>
            <NavItem className="Navbar-Jobs">
              <NavLink to="/profile">Profile</NavLink>
            </NavItem>
            <NavItem className="Navbar-Logout">
              <Link to="/" onClick={logout}>Logout</Link>
            </NavItem>
          ) </>}

          { !user && <>(
              <NavItem className="Navbar-Signup">
                <NavLink to="/signup">Signup</NavLink>
              </NavItem>
              <NavItem className="Navbar-Login">
                <Link to="/login">Login</Link>
              </NavItem>
          )</> }
            {/* <NavItem className="Navbar-Logout">
            <NavLink to="/logout">Log out</NavLink>
            </NavItem> */}
        </Nav>
      </Navbar>
    </div>
  );
}

export default NavBar;
