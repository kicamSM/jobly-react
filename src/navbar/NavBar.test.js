import React from 'react';
import { render, screen } from '@testing-library/react';
import NavBar from './NavBar';
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "../testUtils";
import { Nav } from 'reactstrap';

test("it should render when logged out", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/signup"]}>
        <UserProvider user={null}>
            <NavBar />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(screen.getByText('Jobly')).toBeInTheDocument();
    expect(screen.getByText('Login')).toBeInTheDocument();
    expect(screen.getByText('Signup')).toBeInTheDocument();
  }); 

  it("matches snapshot when logged out", function() {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/signup"]}>
        <UserProvider user={null}>
            <NavBar />
        </UserProvider>
      </MemoryRouter>,
    );
      expect(asFragment()).toMatchSnapshot();
  });
    

test("it should render when logged in", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/signup"]}>
        <UserProvider >
            <NavBar />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(screen.getByText('Jobly')).toBeInTheDocument();
    expect(screen.getByText('Profile')).toBeInTheDocument();
    expect(screen.getByText('Logout testfirst')).toBeInTheDocument();
    expect(screen.getByText('Jobs')).toBeInTheDocument();
    expect(screen.getByText('Companies')).toBeInTheDocument();
  }); 

  it("matches snapshot when logged in", function() {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/signup"]}>
        <UserProvider >
            <NavBar />
        </UserProvider>
      </MemoryRouter>,
    );
      expect(asFragment()).toMatchSnapshot();
  });
