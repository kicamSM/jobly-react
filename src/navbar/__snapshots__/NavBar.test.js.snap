// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
           (
          <li
            class="Navbar-Companies nav-item"
          >
            <a
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="Navbar-Jobs nav-item"
          >
            <a
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="Navbar-Jobs nav-item"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="Navbar-Logout nav-item"
          >
            <a
              href="/"
            >
              Logout testfirst
            </a>
          </li>
          ) 
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="Navbar-Signup nav-item"
          >
            <a
              aria-current="page"
              class="active"
              href="/signup"
            >
              Signup
            </a>
          </li>
          <li
            class="Navbar-Login nav-item"
          >
            <a
              href="/login"
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 2`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
           (
          <li
            class="Navbar-Companies nav-item"
          >
            <a
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="Navbar-Jobs nav-item"
          >
            <a
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="Navbar-Jobs nav-item"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="Navbar-Logout nav-item"
          >
            <a
              href="/"
            >
              Logout testfirst
            </a>
          </li>
          ) 
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
